### PR TITLE
[Headless SXA][NextJS] Support SXA Rendering Variants inside the Placeholder component

### PR DESCRIPTION
**Pull request by illiakovalenko**
## Description / Motivation
* `Placeholder` component will provide required props for SXA Rendering Variants
* `ComponentFactory` will support the selection of the specific component in case you import multiple components from the same file.
* `params.FieldNames` determines that current rendering is SXA rendering variant

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: packages/sitecore-jss-react/src/components/PlaceholderCommon.tsx [hunk 1]
```diff
@@ -197,7 +197,15 @@ export class PlaceholderCommon<T extends PlaceholderProps> extends React.Compone
...
           }),
           ...((placeholderParams || componentRendering.params) && {
-            params: { ...placeholderParams, ...componentRendering.params },
+            params: {
+              ...placeholderParams,
+              ...componentRendering.params,
+              // Provide SXA styles
+              ...(componentRendering.params.FieldNames && {
+                styles: `${componentRendering.params.GridParameters || ''} ${componentRendering
+                  .params.Styles || ''}`,
```
Comment (ambrauer) [inline packages/sitecore-jss-react/src/components/PlaceholderCommon.tsx:206]: Might be good to extract this logic to separate function e.g. `getSxaParams`? There's a bit of SXA magic here.

File: packages/sitecore-jss-react/src/components/PlaceholderCommon.tsx [hunk 2]
```diff
@@ -220,6 +228,14 @@ export class PlaceholderCommon<T extends PlaceholderProps> extends React.Compone
       return null;
     }
 
+    // Render SXA Rendering Variant
+    if (renderingDefinition.params?.FieldNames) {
+      return componentFactory(
```
Comment (ambrauer) [inline packages/sitecore-jss-react/src/components/PlaceholderCommon.tsx:233]: Could we use the same `componentModule` concept we are using for component props service? Would take a bit of rework of those types involved, but the benefit is that those are already being generated in the component factory. E.g. here, we would be able to call something like
```suggestion
      const module = await componentModule(renderingDefinition.componentName);
      return module[renderingDefinition.params.FieldNames];
```

**Review by ambrauer — APPROVED** [review]
Some suggestions (on minor, one potentially larger), otherwise looks great!

File: packages/sitecore-jss-react/src/components/PlaceholderCommon.tsx (hunk 2 above)
Comment (illiakovalenko) [reply packages/sitecore-jss-react/src/components/PlaceholderCommon.tsx:233]: For me - the main issue here is `async` call, and I can't use it inside the `render` function, we can't wait during that process...

File: packages/sitecore-jss-react/src/components/PlaceholderCommon.tsx (hunk 1 above)
Comment (illiakovalenko) [reply packages/sitecore-jss-react/src/components/PlaceholderCommon.tsx:206]: Done

File: packages/sitecore-jss-react/src/components/PlaceholderCommon.tsx (hunk 2 above)
Comment (illiakovalenko) [reply packages/sitecore-jss-react/src/components/PlaceholderCommon.tsx:237]: And `componentModule` basically used for dynamic import, but it's possible to use `componentFactory` to get anything from the file, but synchronously

Comment (ambrauer) [reply packages/sitecore-jss-react/src/components/PlaceholderCommon.tsx:238]: Yeah, the async/sync complicates re-use. Makes sense to me as-is then.

**Pull request merged**